Unable to install via pip on Linux

Installation on Linux fails when installed via
    $ sudo pip install fann2
The output is:
```
Collecting fann2
  Using cached fann2-1.0.7.tar.gz
Installing collected packages: fann2
  Running setup.py install for fann2: started
    Running setup.py install for fann2: finished with status 'error'
    Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-k21ycawc/fann2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-d8aa8m6m-record/install-record.txt --single-version-externally-managed --compile:
    running swig
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/fann2
    copying fann2/__init__.py -> build/lib.linux-x86_64-3.6/fann2
    copying fann2/libfann.py -> build/lib.linux-x86_64-3.6/fann2
    running egg_info
    writing fann2.egg-info/PKG-INFO
    writing dependency_links to fann2.egg-info/dependency_links.txt
    writing top-level names to fann2.egg-info/top_level.txt
    reading manifest file 'fann2.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'fann2.egg-info/SOURCES.txt'
    copying fann2/fann2.i -> build/lib.linux-x86_64-3.6/fann2
    copying fann2/fann2_wrap.cxx -> build/lib.linux-x86_64-3.6/fann2
    copying fann2/fann_cpp_subclass.h -> build/lib.linux-x86_64-3.6/fann2
    running build_ext
    building 'fann2._libfann' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/fann2
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -DSWIG_COMPILE -I./include -I../include -Iinclude -I/usr/include/python3.6m -c fann2/fann2_wrap.cxx -o build/temp.linux-x86_64-3.6/fann2/fann2_wrap.o
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    g++ -pthread -shared -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now build/temp.linux-x86_64-3.6/fann2/fann2_wrap.o -L/usr/lib -ldoublefann -lpython3.6m -o build/lib.linux-x86_64-3.6/fann2/_libfann.cpython-36m-x86_64-linux-gnu.so
    /usr/bin/ld: cannot find -ldoublefann
    collect2: error: ld returned 1 exit status
    error: command 'g++' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-k21ycawc/fann2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-d8aa8m6m-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-k21ycawc/fann2/

```

--- a/ChangeLog
+++ b/ChangeLog
@@ -2,6 +2,12 @@
 CHANGELOG
 =========
 
+1.1.1 - 2017-12-07
+
+  * Added '/usr/lib64' directory to FANN libs search function for setup.py.
+  * Removed unnecessary search block for FreeBSD specifics.
+  * Pylinted setup.py.
+
 1.1.0 - 2017-10-02
 
   * Fixed build for Windows OS.

--- a/setup.py
+++ b/setup.py
@@ -30,7 +30,7 @@ def find_executable(executable, path=None):
     elif sys.platform == 'win32':
         pathext = os.environ['PATHEXT'].lower().split(os.pathsep)
         ext = os.path.splitext(executable)
-        if ext.lower() not in pathext:
+        if ext not in pathext:
             extlist = pathext
     for ext in extlist:
         execname = executable + ext
@@ -68,9 +68,10 @@ def find_fann():
     else:
         dirs = ['/lib', '/usr/lib', '/usr/lib64', '/usr/local/lib', '/usr/pkg/lib']
         for path in dirs:
-            if find_x(path):
-                return True
-        raise Exception("Couldn't find FANN source libs!")
+            if os.path.isdir(path):
+                if find_x(path):
+                    return True
+            raise Exception("Couldn't find FANN source libs!")
 
 def find_swig():
     '''Find SWIG executable path'''
@@ -81,9 +82,9 @@ def find_swig():
 
 def build_swig():
     '''Run SWIG with specified parameters'''
-    print ("Looking for FANN libs...")
+    print("Looking for FANN libs...")
     find_fann()
-    print ("running SWIG...")
+    print("running SWIG...")
     swig_bin = find_swig()
     swig_cmd = [swig_bin, '-c++', '-python', 'fann2/fann2.i']
     subprocess.Popen(swig_cmd).wait()
@@ -111,7 +112,9 @@ setup(
         "License :: OSI Approved :: GNU Lesser General Public License v2 or later (LGPLv2+)",
         "Programming Language :: Python :: 2.7",
         "Programming Language :: Python :: 3.3",
-        "Programming Language :: Python :: 3.4"
+        "Programming Language :: Python :: 3.4",
+        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.6"
     ],
     keywords="ANN artificial intelligence FANN2.2.0 bindings".split(' '),
     zip_safe=False,

--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,7 @@ import subprocess
 from setuptools import setup, Extension, find_packages
 
 NAME = 'fann2'
-VERSION = '1.1.0'
+VERSION = '1.1.1'
 
 with open("README.rst") as f:
     LONG_DESCRIPTION = f.read()
@@ -22,14 +22,14 @@ def find_executable(executable, path=None):
     paths = path.split(os.pathsep)
     extlist = ['']
     if os.name == 'os2':
-        (base, ext) = os.path.splitext(executable)
+        ext = os.path.splitext(executable)
         # executable files on OS/2 can have an arbitrary extension, but
         # .exe is automatically appended if no dot is present in the name
         if not ext:
             executable = executable + ".exe"
     elif sys.platform == 'win32':
         pathext = os.environ['PATHEXT'].lower().split(os.pathsep)
-        (base, ext) = os.path.splitext(executable)
+        ext = os.path.splitext(executable)
         if ext.lower() not in pathext:
             extlist = pathext
     for ext in extlist:
@@ -41,6 +41,7 @@ def find_executable(executable, path=None):
                 fil = os.path.join(pth, execname)
                 if os.path.isfile(fil):
                     return fil
+            break
     else:
         return None
 
@@ -65,10 +66,7 @@ def find_fann():
             return True
         raise Exception("Couldn't find FANN source libs!")
     else:
-        if sys.platform.startswith('freebsd'):
-            dirs = ['/lib', '/usr/lib', '/usr/pkg/lib']
-        else:
-            dirs = ['/lib', '/usr/lib', '/usr/local/lib', '/usr/pkg/lib']
+        dirs = ['/lib', '/usr/lib', '/usr/lib64', '/usr/local/lib', '/usr/pkg/lib']
         for path in dirs:
             if find_x(path):
                 return True

